src/components/legend.ts

```diff
@@ -326,6 +329,38 @@ export module Components {
       }
     }
 
+    /**
+     * Gets the function determining the opacity of the symbols of the Legend.
+     *
+     * @returns {(datum: any, index: number) => number}
+     */
+    public symbolOpacity(): (datum: any, index: number) => number;
```

Comment:
We would need to figure out if we want to always return a function here or maybe union type it between a number and a function